### PR TITLE
feat: add shared folder

### PR DESCRIPTION
**Pull request by njenwei**
### Major
- New package `shared` that is added to `frontend` and `backend`'s tspath. `nest-cli.json` entrypoint modified to handle the new build layout. `dev` and `start` should work correctly.
- Moved `.eslintrc.js` to top level, and shared across all sub-packages
- Added new eslint plugin `plugin:@typescript-eslint/recommended-requiring-type-checking`
- Update Dockerfile to build smaller packages - backend nest server now configured to serve frontend build package.
- Update backend sequalise database to use config specified. It now points to the postgres docker in dev mode. 
- Backend uses `dotenv` to load environment variables, we follow the loading precedence set by https://github.com/entropitor/dotenv-cli/issues/37

### Minor
- Added scripts to download env vars from SSM. Added scripts to generate cli commands to put variables in SSM
- Reformat imports
- Upgrade remaining deps and move `eslint` plugins to top level `package.json`
- Removed `travis` and `gitpod`

### New Deps
- `@nestjs/serve-static`
- `dotenv`
- `pg`
- `@aws-sdk/client-ssm`
- `dotenv-cli`
- `react-app-alias-ex`
- `eslint-plugin-storybook`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-RUN npm run on-backend -- build && npm run on-frontend -- build
+RUN npm ci
+RUN ENV=$ENV REACT_APP_ENV=$ENV npm run build \
+  && npm prune --production
 
 FROM node:lts-alpine
 WORKDIR /usr/src/app
+
+COPY --from=node-modules-builder /usr/src/app/backend ./backend
+COPY --from=node-modules-builder /usr/src/app/frontend/build ./frontend/build
```
Comment (LoneRifle) [inline Dockerfile:16]: I take it that we only copy `/build` because webpack bundles everything?

File: frontend/package.json [hunk 2]
```diff
@@ -35,24 +35,23 @@
...
-    "react-app-rewire-alias": "^1.1.7",
+    "react-app-alias-ex": "^2.1.0",
     "react-app-rewired": "^2.2.1",
     "react-scripts": "^5.0.1",
     "typescript": "^4.7.4",
     "webpack": "^5.74.0"
   },
+  "overrides": {
+    "react-refresh": "0.14.0"
+  },
```
Comment (LoneRifle) [inline frontend/package.json:54]: ?

File: shared/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  extends: ['./../.eslintrc.js'],
```
Comment (LoneRifle) [inline shared/.eslintrc.js:2]: nit
```suggestion
  extends: ['../.eslintrc.js'],
```

**Review by LoneRifle — APPROVED** [review]
lgtm. Good work on rejigging for the shared dir!

File: frontend/package.json (hunk 2 above)
Comment (njenwei) [reply frontend/package.json:54]: Storybook and react-script conflict issue - it tries to pull both v0.11.0 and v0.14.0 which causes the error `Relative imports outside of src/ are not supported.` to be thrown during dev. This fixes it by pinning the versions internally.

- https://github.com/facebook/create-react-app/issues/11810

```
# not override
$ npm ls react-refresh
client@0.1.0 /*******/ts-template/frontend
├─┬ @storybook/preset-create-react-app@4.1.2
│ └─┬ @pmmmwh/react-refresh-webpack-plugin@0.5.7
│   └── react-refresh@0.14.0
├─┬ @storybook/react@6.5.9
│ └── react-refresh@0.11.0
└─┬ react-scripts@5.0.1
  └── react-refresh@0.11.0

# override
$ npm ls react-refresh
client@0.1.0 /*******/ts-template/frontend
├─┬ @storybook/preset-create-react-app@4.1.2
│ └─┬ @pmmmwh/react-refresh-webpack-plugin@0.5.7
│   └── react-refresh@0.14.0 deduped
├─┬ @storybook/react@6.5.9
│ └── react-refresh@0.14.0
└─┬ react-scripts@5.0.1
  └── react-refresh@0.14.0 deduped
```

File: Dockerfile (hunk 1 above)
Comment (njenwei) [reply Dockerfile:16]: Yup

**Pull request merged**